packages/database/src/migrations/20210819150000-AddCovidFiji7DayPositivityRateReport.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+import {
+  insertObject,
+  generateId,
+  findSingleRecord,
+  findSingleRecordBySql,
+  deleteObject,
+} from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const generateConfig = () => ({
+  code: `FJ_Covid_7_Day_Positivity_Rate`,
+  reportConfig: {
+    fetch: {
+      aggregations: [
+        {
+          type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+          config: {
+            dataSourceEntityType: 'facility',
+            aggregationEntityType: 'requested',
+          },
+        },
+      ],
+      dataElements: ['COVID_FJ_7_Day_Rolling_Pos_Tests', 'COVID_FJ_7_Day_Rolling_Num_Tests'],
+    },
+
+    transform: [
+      'keyValueByDataElementName',
+      {
+        transform: 'aggregate',
+        period: 'group',
+        '...': 'drop',
+        COVID_FJ_7_Day_Rolling_Num_Tests: 'sum',
+        COVID_FJ_7_Day_Rolling_Pos_Tests: 'sum',
+      },
+      {
+        transform: 'filter',
+        where:
+          'exists($row.COVID_FJ_7_Day_Rolling_Num_Tests) and $row.COVID_FJ_7_Day_Rolling_Num_Tests > 0',
```

Comment:
Is this filter on `$row.COVID_FJ_7_Day_Rolling_Num_Tests > 0` to avoid division by 0? If so, yeah this can be rather annoying 😞 

Often times I achieve this by doing a ternary in the select instead, eg:
```
        transform: 'select',
        "'value'": ' $row.COVID_FJ_7_Day_Rolling_Num_Tests == 0 ? 0 : $row.COVID_FJ_7_Day_Rolling_Pos_Tests / $row.COVID_FJ_7_Day_Rolling_Num_Tests',
```

The reason for this is to still display data when the value is actually meant to be 0 (eg. a known 0 tests performed that day should have 0 value, not 'no data').

Still happy to keep the filter for `exists($row.COVID_FJ_7_Day_Rolling_Num_Tests)` though, however I'm guessing that should never actually happen?

Reply:
Hmmm, yeah that's a good point! Although I'm not sure if 0% positivity makes sense if there are no tests from a project point of view, I'll ask on the card :)).

Reply:
I think the filter can stay based on this comment: https://linear.app/bes/issue/SOL-296#comment-4a7b6553?
Let me know if you're happy with that 👍

Reply:
Yup, makes sense to me! Good call 👍